citizennet/purescript-kafkajs
# FBCM-5195 Setup Spago Project

## What does this pull request do?

Initialize the PS project with `spago` and `purs-tidy` and add a `Makefile`. Add `kafkajs` repo as a submodule so that we can use the docker compose file that `kafkajs` uses to test our own code. Add `Test.Main` that automatically prepares Zookeeper and Kafka in the environment ready to connect to producers/consumers. Add documentation on basic shell commands for people new to Kafka to play with.

## How should this be manually tested?

- [ ] no `nix` set up yet so you need `docker` and `docker-compose` [installed](https://docs.docker.com/compose/install/)
- [ ] `make build`
  - should download `npm` and `spago` dependencies and build PS code
  - should start Zookeeper and Kafka, wait for brokers `kafka1`, `kafka2`, `kafka3` to be ready, and then stop Zookeeper and Kafka
- [ ] (optional) try out commands under `docs/shell-commands.md`
